lib/ansible/module_utils/urls.py

```diff
@@ -1049,11 +1049,36 @@ def fetch_url(module, url, data=None, headers=None, method=None,
                      url_password=password, http_agent=http_agent, force_basic_auth=force_basic_auth,
                      follow_redirects=follow_redirects, client_cert=client_cert,
                      client_key=client_key, cookies=cookies)
-        info.update(r.info())
+        # Lowercase keys, to conform to py2 behavior, so that py3 and py2 are predictable
+        info.update(dict((k.lower(), v) for k, v in r.info().items()))
```

Comment:
This looks like a bug fix. If that's the case it should go into a separate PR, preferably along with tests.

Also, doesn't this cause potential issues with backwards compatibility?

Reply:
I suppose it could be considered a bug fix.

While writing tests I noticed that py2 gives lowercased names, while py3 gives capitalization of the name from the response.

I can't see how it would cause a potential backwards compat issue. I went with the py2 default, because all of our modules should support py2. The only way it could cause a backwards compat issue, is if a module was calling `fetch_url` and only supported python3.  If they supported both py2 and py3, then the code would have to already handle lowercase and mixed case, so it should work as expected.

This normalizes/standardizes return values from this method.

Reply:
Also, in general, this whole PR is a bug fix.  Largely due to difference in behavior of python3 vs python2.  The goal is to normalize the functionality.

The only place not a bugfix is the addition of `cookies_string`.  That was added to ease issues related to the original bug report of missing duplicate headers.